src/layers.c

```diff
@@ -131,60 +137,62 @@ surface_commit_notify(struct wl_listener *listener, void *data)
 		return;
 	}
 
-	if (layer_surface->current.committed
-			|| layer->mapped != layer_surface->mapped) {
-		layer->mapped = layer_surface->mapped;
-		struct output *output =
-			output_from_wlr_output(layer->server, wlr_output);
-		layers_arrange(output);
+	/*
+	 * 'committed' here refers to change in layer-surface state including
+	 * desired-size, anchor, margin, exlusive-zone, keyboard-interactivity
+	 * and/or change-layer, rather than frame-commit.
```

Comment:
I don't think this comment is correct, I added a `wlr_log(WLR_ERROR, "layer commit");` after the comment and it gets triggered on frame updates as well. Can be seen by just having some panel running and then switching focus between two windows.

Reply:
Comment removed.